### PR TITLE
CDAP-6528 Attempt Macro substituion only if the value is provided by user.

### PR DESCRIPTION
**Pull request by sagarkapare**
https://issues.cask.co/browse/CDAP-6528
This should fix the latest ITN failures in the BatchCubeSink.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java [hunk 1]
```diff
@@ -216,8 +217,11 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     Map<String, PluginPropertyField> pluginPropertyFieldMap = plugin.getPluginClass().getProperties();
     MacroParser macroParser = new MacroParser(macroEvaluator);
     for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
+      String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
+      if (Strings.isNullOrEmpty(macroValue)) {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java]: Why? The value cannot be null or empty??

**Comment by knazir** [comment]
I addressed this in this PR: https://github.com/caskdata/cdap/pull/6063 but it was taking longer to merge than I thought. I'll merge your change into my branch when mine is finished being reviewed.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java [hunk 2]
```diff
@@ -216,8 +217,11 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     Map<String, PluginPropertyField> pluginPropertyFieldMap = plugin.getPluginClass().getProperties();
     MacroParser macroParser = new MacroParser(macroEvaluator);
     for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
+      String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java]: this is a misleading name. Lets name it propertyValue

**Comment by albertshau** [comment]
lgtm

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java (hunk 1 above)
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java]: And if the field is not macro supported, you still skip it? Seems not right to me.

**Comment by sagarkapare** [comment]
@albertshau @chtyim Please take a look again. Thanks!

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT4431-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java [hunk 3]
```diff
@@ -215,12 +215,13 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
...
     Map<String, PluginPropertyField> pluginPropertyFieldMap = plugin.getPluginClass().getProperties();
     MacroParser macroParser = new MacroParser(macroEvaluator);
-    for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
-      if (pluginEntry.getValue().isMacroSupported()) {
-        String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
-        properties.put(pluginEntry.getKey(), macroParser.parse(macroValue));
+
+    for (Map.Entry<String, String> property : plugin.getProperties().getProperties().entrySet()) {
+      PluginPropertyField field = pluginPropertyFieldMap.get(property.getKey());
+      if (field.isMacroSupported()) {
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java]: its possible field is null if the user provided a property that is not in the PluginConfig. so this should check that field is not null and field is macro supported.

**Comment by albertshau** [comment]
one comment, then lgtm

**Comment by sagarkapare** [comment]
Build is green here - http://builds.cask.co/browse/CDAP-DUT4431-2
Squashing and merging.

**Pull request merged**